packages/@uppy/aws-s3-multipart/src/MultipartUploader.js

```diff
@@ -105,6 +108,12 @@ class MultipartUploader {
           onComplete: this.#onPartComplete(j),
           shouldUseMultipart,
         }
+        if (this.#isRestoring) {
+          const size = i + chunkSize > fileSize ? fileSize - i : chunkSize
+          this.#chunks[j].setAsUploaded = () => {
```

Comment:
why a function and not setting `uploaded` directly, since we use `size` from this scope anyway?

Reply:
We don't know yet if it was already uploaded

Reply:
could use a comment how this is used later